src/ORM/DataObject.php

```diff
@@ -2386,7 +2386,8 @@ public function getDefaultSearchContext()
      */
     public function getGeneralSearchFieldName(): string
     {
-        return $this->config()->get('general_search_field_name');
+        $configSetting = $this->config()->get('general_search_field_name');
+        return is_string($configSetting) ? $configSetting : self::$general_search_field_name;
```

Comment:
Is this ever hit before cache is generated? If not I'd be inclined to leave it as is.

Reply:
I'm inclined to disagree on principle.
Or, if it is to stay 'as is' then the return type should be dropped.

The change is small, does no harm, and enforces the return type.
I guess it could lead to an obscure error later in execution by silently ignoring a bad config setting, perhaps throwing an error would be better?

Reply:
PHP itself will already throw an error if the type is wrong, which will tell developers they've set the config to an invalid value.
My hesitance here comes from the fact that we're not really doing this anywhere else, except perhaps in places where different value types are expected and _need_ to be handled.

This seems like the sort of thing we should have a considered standard approach for, and have an issue to handle these across the board in a uniform way. 

As it stands, I'd prefer this pr to only include the changes for the same site issue -but if you want to keep them here it's be good to have another opinion or two.

Reply:
I can remove the non samesite things, but since it's not super urgent lets get some other opinions in.

I agree, [I think there should be a standard approach](https://github.com/silverstripe/silverstripe-framework/issues/10738) - no points awarded for guessing my position :P

Reply:
Please remove the non samesite things, it's not related to the issue being fixed and just makes for muddled PRs